src/api/spec/controllers/webui/user_controller_spec.rb

```diff
@@ -271,8 +271,15 @@
     skip
   end
 
-  describe "GET #admin" do
-    skip
+  describe "POST #admin" do
+    before do
+      login admin_user
+      post :admin, params: { user: user.login }
+    end
+
+    it 'applies the Admin role properly' do
+      expect(user.roles.find_by(title: 'Admin')).to_not be_nil
```

Comment:
This is not testing what we want as the `admin_user` also has the admin role, and we want to check if the admin role was assigned to `user`. :disappointed:  You have to test that `user` is included in `user.roles.find_by(title: 'Admin')` :wink:

Reply:
user.roles holds all roles a user has. not every role. this is a relation.

Reply:
I read it wrong :wink: